nft_ingester/src/api/backfilling_state_consistency.rs

```diff
@@ -33,36 +29,29 @@ impl BackfillingStateConsistencyChecker {
 
     pub(crate) async fn run(
         &self,
-        tasks: Arc<Mutex<JoinSet<Result<(), JoinError>>>>,
-        rx: tokio::sync::broadcast::Receiver<()>,
+        cancellation_token: CancellationToken,
         rocks_db: Arc<Storage>,
         _consistence_backfilling_slots_threshold: u64,
     ) {
         for asset_type in ASSET_TYPES {
             let _rocks_db = rocks_db.clone();
-            let mut rx = rx.resubscribe();
             let _overwhelm_backfill_gap = match asset_type {
                 AssetType::NonFungible => self.overwhelm_nft_backfill_gap.clone(),
                 AssetType::Fungible => self.overwhelm_fungible_backfill_gap.clone(),
             };
-            tasks.lock().await.spawn(async move {
-            while rx.is_empty() {
-                // TODO: refactor this to use parameter from storage and last slot from slot storage
-                // overwhelm_backfill_gap.store(
-                //     rocks_db.bubblegum_slots.iter_start().count().saturating_add(rocks_db.ingestable_slots.iter_start().count())
-                //         >= consistence_backfilling_slots_threshold as usize,
-                //     Ordering::Relaxed,
-                // );
-                tokio::select! {
-                    _ = tokio::time::sleep(Duration::from_secs(CATCH_UP_SEQUENCES_TIMEOUT_SEC)) => {},
-                    _ = rx.recv() => {
-                        info!("Received stop signal, stopping BackfillingStateConsistencyChecker...");
-                        return Ok(());
-                    }
+            let cancellation_token = cancellation_token.child_token();
+            usecase::executor::spawn(async move {
+                while !cancellation_token.is_cancelled() {
+                    // TODO: refactor this to use parameter from storage and last slot from slot storage
+                    // overwhelm_backfill_gap.store(
+                    //     rocks_db.bubblegum_slots.iter_start().count().saturating_add(rocks_db.ingestable_slots.iter_start().count())
+                    //         >= consistence_backfilling_slots_threshold as usize,
+                    //     Ordering::Relaxed,
+                    // );
+                    //
+                    tokio::time::sleep(Duration::from_secs(CATCH_UP_SEQUENCES_TIMEOUT_SEC)).await;
```

Comment:
why not keep the tokyo select and just replace `rx.recv()` with `cancellation_token.cancelled()` in such cases?

Reply:
good catch, though i will replace with `CancellationToken::run_until_cancelled`